pkg/stanza/fileconsumer/design/design.md

```diff
@@ -0,0 +1,279 @@
+# File Matching
```

Comment:
How much of this is duplicated from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/operator/input/file/design.md?